internal/target/repository.go

```diff
@@ -32,14 +29,15 @@ type Repository struct {
 // NewRepository creates a new target Repository. Supports the options: WithLimit
 // which sets a default limit on results returned by repo operations.
 func NewRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*Repository, error) {
+	const op = "target.NewRepository"
 	if r == nil {
-		return nil, stderrors.New("error creating db repository with nil reader")
+		return nil, errors.New(errors.InvalidParameter, op, "nil reader")
```

Comment:
Is there a nil error code we can return instead? Feels like it is used a lot 🤔

Reply:
`InvalidParameter` is the generic type used for all these parameter checks, we don't really care at the `Code` level if it was nil, empty, set when it should not have been set, or some other state.  What we care about is that the parameter sent to the function was invalid.